### PR TITLE
feat: don't create PR when contributor already have contribution type

### LINKED ISSUE #345: Don’t create PR when contributor is already exist
**Is your feature request related to a problem? Please describe.**
When a contribution been made by returning contributor and we ask a bot to add the contributor to the list it create an empty PR.

**Describe the solution you'd like**
What if when the bot is been asked to add a contributor it will check if the contributor is missing and only then it will add him, otherwise, it will replay with custom replay message or the default one “The user with this contribution type is already exist”

**Flow:**
Flow

### PR DESCRIPTION
**Pull request by rluvaton**
Closes #345

### CODE REVIEW

File: lib/modules/config.js [hunk 1]
```diff
@@ -90,25 +90,34 @@ class Config {
     return this.fileSha;
   }
 
-  async addContributor({ login, contributions, name, avatar_url, profile }) {
+  #findOldContributions(username) {
```
Comment (gr2m) [inline lib/modules/config.js:93]: what's the `#` sign? Is this valid JavaScript? I've never seen that before

Comment (rluvaton) [reply lib/modules/config.js:93]: > what's the `#` sign? Is this valid JavaScript? I've never seen that before

It's private class field at run time.

[MDN - Private Class Fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields)

**Comment by gr2m** [comment]
@rluvaton I've deployed your update as preview to Vercel and updated https://github.com/apps/all-contributors-preview to point to that deployment. Can you install https://github.com/apps/all-contributors-preview on a test repository and see if it resolves your problem? Make sure that you don't have both https://github.com/apps/allcontributors and https://github.com/apps/all-contributors-preview installed on the same repository, otherwise the two will conflict with each other

**Comment by rluvaton** [comment]
@gr2m I installed it and it doesn't respond to any of my comments.
Is there a way to debug it?

**Comment by gr2m** [comment]
Here is what I see in the logs

image

It might be that the private class field is not supported by the node version used by Vercel

**Comment by rluvaton** [comment]
Thank you for your help, I fixed it

**Comment by gr2m** [comment]
can you try again

**Comment by rluvaton** [comment]
@gr2m Work like a charm!

> My tests: https://github.com/rluvaton/all-contributor-playground/pull/5

**Review by gr2m — APPROVED** [review]
Great work, thank you 👏🏼

**Comment by rluvaton** [comment]
@gr2m Don't forget to add me as a contributor 😉 😉

**Comment by gr2m** [comment]
haha I just waited for the new deployment to finish so I could test the changes in production :D 

@all-contributors please add @rluvaton for code, bug

**Comment by ouuan** [comment]
I used to use the old behavior to update the README. Now how do I update README without adding a new contributor/contribution type?

**Comment by rluvaton** [comment]
I'm having a hard time to understand 😬, what do you mean by:

> I used to use the old behavior to update the README.

Update with what changes?

**Comment by ouuan** [comment]
> Update with what changes?

For example, manual changes of `.all-contributorsrc`.

**Comment by ouuan** [comment]
It was a workaround of #157.

**Comment by rluvaton** [comment]
And you want your manual change will update the README?

**Comment by ouuan** [comment]
I think the best solution is to implement what #157 requests.

**Comment by rluvaton** [comment]
I see, I'm a bit busy at the moment so I won't be able to create a PR, but you're more than welcome 🤗 

From what I understand from the code, you need to add the action here:

https://github.com/all-contributors/app/blob/00d277fec6dad3b0cbe097f8d9faa446e8f380cf/lib/process-issue-comment.js#L30

Extract the code after to an `add` function 

Add the action to here:
https://github.com/all-contributors/app/blob/00d277fec6dad3b0cbe097f8d9faa446e8f380cf/lib/parse-comment.js#L197

Take some of the code from here:

https://github.com/all-contributors/app/blob/00d277fec6dad3b0cbe097f8d9faa446e8f380cf/lib/add-contributor.js#L39

And I think that's it 😅

**Pull request merged**